src/timing.hpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The clvk authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+
+#include "log.hpp"
+
+class cvk_timer {
+public:
+    cvk_timer(std::string name) : m_name(name) {}
+
+#ifdef CVK_ENABLE_TIMING
+
+    ~cvk_timer() {
+        double average = m_count ? (m_total_time / 1.0e6) / m_count : 0.0;
+        cvk_warn("%.2f ms -> %s (%d blocks, avg %.3f ms)", m_total_time / 1.0e6,
```

Comment:
We could maybe assign a logging group to these messages. They're all printed at the end though so not critical at all to be able to filter them out.